python/lsst/ctrl/mpexec/preExecInit.py

```diff
@@ -49,22 +50,50 @@ class PreExecInit:
         Data butler instance.
     taskFactory : `~lsst.pipe.base.TaskFactory`
         Task factory.
-    skipExisting : `bool`, optional
+    extendRun : `bool`, optional
         If `True` then do not try to overwrite any datasets that might exist
         in ``butler.run``; instead compare them when appropriate/possible.  If
         `False`, then any existing conflicting dataset will cause a butler
         exception to be raised.
     """
-    def __init__(self, butler, taskFactory, skipExisting=False):
+    def __init__(self, butler, taskFactory, extendRun=False):
         self.butler = butler
         self.taskFactory = taskFactory
-        self.skipExisting = skipExisting
-        if self.skipExisting and self.butler.run is None:
+        self.extendRun = extendRun
+        if self.extendRun and self.butler.run is None:
             raise RuntimeError(
-                "Cannot perform skipExisting logic unless butler is initialized "
+                "Cannot perform extendRun logic unless butler is initialized "
                 "with a default output RUN collection."
             )
 
+    @classmethod
+    def datasetTypeNames(cls, taskDefs: Iterable[TaskDef]) -> Iterator[str]:
+        """Return the names of all dataset types for a given TaskDefs.
+
+        Parameters
+        ----------
+        taskDefs : iterable of `TaskDef`
+            TaskDefs constituting a pipeline.
+
+        Yields
+        ------
+        datasetTypeName : `str`
+            Name of the dataset type.
+        """
+        # Package versions dataset type
+        yield "packages"
```

Comment:
I wanted this method here mostly because it's PreExecInit which determines what datasets are actually stored. pipe_base of course knows about dataset types and I should probably use PipelineDatasetTypes method here to make list (similarly to initializeDatasetTypes method in this class) but PipelineDatasetTypes does a lot more that I need here (but I'll probably switch anyways).

I agree  "packages" should come from pipe_base. I don't want to add it to TaskDef as that class is for task-specific things, but PipelineDatasetTypes is probably a good place for it.